File: sponge-standalone/examples/standalone/error_reporting_function/error_reporting_function.py
"""
Sponge Knowledge base
Error reporting
"""

class HelloWorldTrigger(Trigger):
    def onConfigure(self):
        self.event = "helloEvent"
    def onRun(self, event):
        print event.get("say")

def onStartup():
    whatIsThis.doSomething()
    sponge.event("helloEvent").set("say", "Hello World!").send()

File: sponge-standalone/examples/standalone/error_reporting_function/error_reporting_function.js
/**
 * Sponge Knowledge base 
 * Error reporting
 */

var HelloWorldTrigger = Java.extend(Trigger, {
    onConfigure: function(self) {
        self.event = "helloEvent";
    },
    onRun : function(self, event) {
        print(event.get("say"));
    }
});

function onStartup() {
    whatIsThis.doSomething();
    sponge.event("helloEvent").set("say", "Hello World!").send();
}
